File: sol/hello_world.f90
program hello

implicit none

include "mpif.h"

integer :: rank, comm_size, ierr

call MPI_Init(ierr)
call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)
call MPI_Comm_size(MPI_COMM_WORLD, comm_size, ierr)

print *, "Hello from rank ", rank, " of ", comm_size

call MPI_Finalize(ierr)

end program

File: lab1/sol/hello_world.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char *argv[]) {

  int myrank, size;

  MPI_Init(&argc, &argv);                 /* Initialize MPI       */
  MPI_Comm_rank(MPI_COMM_WORLD, &myrank); /* Get my rank          */
  MPI_Comm_size(MPI_COMM_WORLD, &size);   /* Get the total
					     number of processors */
  printf("Processor %d of %d: Hello World\n", myrank, size);

  MPI_Finalize();                         /* Terminate MPI        */

}

